src/guiClasses/MainPanelGUI.java

```diff
@@ -2,86 +2,92 @@
 
 import main.UniversitiesInformation;
 import main.University;
-import main.fileInput;
 
 import javax.swing.*;
+import java.awt.*;
 import java.awt.event.ActionEvent;
 import java.awt.event.ActionListener;
+import java.awt.event.AdjustmentEvent;
+import java.awt.event.AdjustmentListener;
 import java.util.ArrayList;
 
-
-public class MainPanelGUI extends JFrame implements ActionListener {
-    public static JLabel title = new JLabel("All Programs");
-    public static JButton nextBtn = new JButton();
-    public static JButton backBtn = new JButton();
-    public static JComboBox<String> combobox1 = new JComboBox<>();
-    public static String[] names = fileInput.names;
-    public static double[] overallAverages = fileInput.intOverallAverages;
-    public static String[] namesSortedAverage = names;
-    public static String[] arrayToUse = names;
-    public static int currentIndex = 0;
-    public static int currentPage = 0;
-    public static boolean reversed = false;
-    String[] sorting = new String[]{"A-Z", "Z-A", "Average: Low to High", "Average: High to Low"};
-    int index = 0;
-    String info = "insert info about uni dashodiahsdoiashdoiashdas";
-    JPanel uniPanel;
-    JComboBox<String> combobox2 = new JComboBox<>();
-    public static int numCards;
+public class MainPanelGUI extends JFrame implements ActionListener, AdjustmentListener{
```

Comment:
Pretty sure Jordan requested for the screens to be done on a panel. Naming the class MainPanelGUI doesn't really count since you're still extending JFrame.